src/concern/mod.rs

```diff
@@ -39,34 +90,35 @@ pub enum ReadConcern {
     Custom(String),
 }
 
-impl PartialEq for ReadConcern {
```

Comment:
given that this `ParitalEq` implementation led to a bug in our URI parser, I wasn't inclined to keep it around. It may be mildly confusing for a user to see that `ReadConcernLevel::Custom("majority") != ReadConcernLevel::Majority`, but that should be rare given that `ReadConcern::custom` parses correct variant out.